app/delivery/send_to_providers.py

```diff
@@ -138,16 +138,22 @@ def send_email_to_provider(notification: Notification):
     template_dict = dao_get_template_by_id(notification.template_id, notification.template_version).__dict__
     plain_text_email = PlainTextEmailTemplate(template_dict, values=personalisation_data)
 
-    html_content = notification.template.html
-    if html_content:
-        html_content = html_content.replace('xx_notification_id_xx', str(notification.id))
-        for key, value in personalisation_data.items():
-            # Escape the key to prevent regex injection
-            key = re.escape(key)
-            # Match the placeholder in HTML. The regex captures the placeholder and any surrounding whitespace.
-            # The span tag is dictated by the Field class in the utils template.
-            regex = rf"<span class='placeholder'>\s*\(\(\s*{key}\s*\)\)\s*</span>"
-            html_content = re.sub(regex, str(value), html_content)
+    if is_feature_enabled(FeatureFlag.STORE_TEMPLATE_CONTENT):
+        html_content = notification.template.html
+        if html_content:
+            html_content = html_content.replace('xx_notification_id_xx', str(notification.id))
+            for key, value in personalisation_data.items():
+                # Escape the key to prevent regex injection
+                key = re.escape(key)
+                # Match the placeholder in HTML. The regex captures the placeholder and any surrounding whitespace.
+                # The span tag is dictated by the Field class in the utils template.
+                regex = rf"<span class='placeholder'>\s*\(\(\s*{key}\s*\)\)\s*</span>"
+                html_content = re.sub(regex, str(value), html_content)
+    else:
+        html_email = HTMLEmailTemplate(
+            template_dict, values=personalisation_data, **get_html_email_options(notification.template, notification.id)
```

Comment:
`notification.id` is a UUID instance.  The function expects a string.
```suggestion
            template_dict, values=personalisation_data, **get_html_email_options(notification.template, str(notification.id))
```